src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/InitializationService.java

```diff
@@ -0,0 +1,315 @@
+        /* get serverconf from db if exists
+           - this is for partial init support since no server confs should yet exist
+         */
+        ServerConfType serverConf = serverConfService.getOrCreateServerConf();
+        ClientType ownerClient = null;
+        // get client from db if exists - this is for partial init support since no clients should yet exist
+        Optional<ClientType> foundClient = serverConf.getClient().stream()
+                .filter(clientType -> clientType.getIdentifier().equals(ownerClientId))
+                .findFirst();
+        ownerClient = foundClient.orElse(getInitialClient(ownerClientId));
+        ownerClient.setConf(serverConf);
+        // again for partial init support: if the client already exists there is no reason to add it again
+        if (!serverConf.getClient().contains(ownerClient)) {
+            serverConf.getClient().add(ownerClient);
+        }
+        serverConf.setOwner(ownerClient);
+        serverConf.setServerCode(securityServerCode);
+        return serverConf;
+    }
+
+    /**
+     * Helper to check for warnings
+     * @param ownerClientId
+     * @param securityServerCode
+     * @throws UnhandledWarningsException
+     */
+    private void checkForWarnings(ClientId ownerClientId, String securityServerCode)
+            throws UnhandledWarningsException {
+        String ownerMemberName = globalConfFacade.getMemberName(ownerClientId);
+        SecurityServerId serverId = SecurityServerId.create(ownerClientId, securityServerCode);
+        List<WarningDeviation> warnings = new ArrayList<>();
+        if (StringUtils.isEmpty(ownerMemberName)) {
+            WarningDeviation memberWarning = new WarningDeviation(WARNING_INIT_UNREGISTERED_MEMBER,
+                    ownerClientId.toShortString());
+            warnings.add(memberWarning);
+        }
+        if (globalConfFacade.existsSecurityServer(serverId)) {
+            WarningDeviation memberWarning = new WarningDeviation(WARNING_INIT_SERVER_ID_EXISTS,
+                    serverId.toShortString());
+            warnings.add(memberWarning);
+        }
+        if (!warnings.isEmpty()) {
+            throw new UnhandledWarningsException(warnings);
+        }
+    }
+
+    /**
+     * Helper to create an initial client
+     * @param clientId
+     * @return
+     */
+    private ClientType getInitialClient(ClientId clientId) {
+        ClientType localClient = clientService.getLocalClient(clientId);
+        if (localClient == null) {
+            localClient = new ClientType();
+            localClient.setIdentifier(clientId);
+            localClient.setClientStatus(ClientType.STATUS_SAVED);
+            localClient.setIsAuthentication(IsAuthentication.SSLAUTH.name());
+        }
+        return localClient;
+    }
+
+    /**
+     * Verify that the initialization process can proceed and that the security server has not already been
+     * initialized. This means verifying that an anchor has been imported,
```

Comment:
This method does not verify that anchor has been imported, but javadoc days it does?

Reply:
The anchor is verified before this. The whole prerequisite check is now different - so fixed